apps/prairielearn/src/pages/publicAssessmentQuestions/publicAssessmentQuestions.ts

```diff
@@ -0,0 +1,51 @@
+import * as express from 'express';
+import asyncHandler from 'express-async-handler';
+
+import * as error from '@prairielearn/error';
+
+import { selectAssessmentQuestions } from '../../models/assessment-question.js';
+import { checkAssessmentPublic, selectAssessmentById } from '../../models/assessment.js';
+import { selectCourseByCourseInstanceId } from '../../models/course.js';
+
+import { PublicAssessmentQuestions } from './publicAssessmentQuestions.html.js';
+
+const router = express.Router({ mergeParams: true });
+
+router.get(
+  '/',
+  asyncHandler(async (req, res) => {
+    const assessment_id = req.params.assessment_id;
+    const course_instance_id = req.params.course_instance_id;
+    const isAssessmentPublic = await checkAssessmentPublic(assessment_id);
+    if (!isAssessmentPublic) {
+      throw new error.HttpStatusError(404, 'Not Found');
+    }
+
+    const course = await selectCourseByCourseInstanceId(course_instance_id.toString());
+    res.locals.course = course;
+    res.locals.assessment = await selectAssessmentById(assessment_id);
+
+    const questions = await selectAssessmentQuestions({
+      assessment_id,
+      course_id: course.id,
+    });
+
+    // Filter out non-public assessments
+    for (const question of questions) {
+      question.other_assessments =
+        question.other_assessments?.filter((assessment) => assessment.share_source_publicly) ?? [];
+    }
+
+    res.send(
+      PublicAssessmentQuestions({
+        resLocals: res.locals,
+        assessment: res.locals.assessment,
```

Comment:
Nit: I'd prefer it if we don't round-trip `assessments` through `res.locals`, as we lose type safety when we do that.

If we want to keep the tabs on this page, it seems like the tabs will still need access to `assessment` via `res.locals`, in which case we can add it to `res.locals` as well.

Reply:
done in 3a35b9749e6b83547eb14988f73709fa2d9075a5